### PR TITLE
Update

### PR DESCRIPTION
**Pull request by AustinShalit**
Update all build dependancies to latest

### CODE REVIEW

**Comment by AustinShalit** [comment]
Retest this please

File: build.gradle.kts [hunk 1]
```diff
@@ -211,16 +211,16 @@ publishing {
...
-        return WPILibVersion.version
-    } else {
-        return fallback
+    return when {
+        project.hasProperty("publishVersion") -> {
+            val publishVersion: String by project
+            publishVersion
+        }
+        WPILibVersion.version != "" -> WPILibVersion.version
+        else -> fallback
```
Comment (PeterJohnson) [inline build.gradle.kts]: This feels a lot harder to read.  Is this change necessary?
